apps/web-app/components/settings/addNewGroup.vue

```diff
@@ -4,47 +4,264 @@
+  const newIdentitySendNameValid = ref<boolean | 'remote' | null>(null);
+  const newIdentityCatchAll = ref(false);
+  const newDomainNameValid = ref<boolean | 'remote' | null>(null);
+  const createEmailIdentityForGroup = ref(false);
+
+  watchDebounced(
+    newGroupNameValue,
+    async () => {
+      if (
+        newIdentitySendNameTempValue.value === newIdentitySendNameValue.value
+      ) {
+        const newValue = newGroupNameValue.value
+          .toLowerCase()
+          .replace(/[^a-z0-9]/g, '');
+        newIdentitySendNameValue.value = newValue;
+        newIdentitySendNameTempValue.value = newValue;
+      }
+    },
+    {
+      debounce: 500,
+      maxWait: 5000
+    }
+  );
+
+  watchDebounced(
+    newIdentitySendNameValue,
+    async () => {
+      if (newIdentitySendNameValid.value === 'remote') {
+        const { available, error } =
+          await $trpc.org.crud.checkSlugAvailability.query({
+            slug: newIdentitySendNameValue.value
+          });
+        if (!available) {
+          newIdentitySendNameValid.value = false;
+          newIdentitySendNameValidationMessage.value = 'Not available';
+        }
+        available && (newIdentitySendNameValid.value = true);
+      }
+    },
+    {
+      debounce: 500,
+      maxWait: 5000
+    }
+  );
+
+  interface OrgDomains {
+    domainPublicId: string;
+    domain: string;
+  }
+  const selectedDomain = ref<OrgDomains | undefined>(undefined);
+
+  const orgDomains = ref<OrgDomains[]>([]);
+
+  const { data: orgDomainsData, pending: orgDomainsPending } =
+    await $trpc.org.mail.domains.getOrgDomains.useLazyQuery(
+      {},
+      { server: false }
+    );
+
+  interface OrgMembers {
+    publicId: String;
+    name: String;
+    handle: String;
+    title: String | null;
+    keywords: String;
+  }
+  const selectedOrgMembers = ref<OrgMembers[]>([]);
+
+  interface OrgUserGroups {
+    publicId: String;
+    name: String;
+    description: String | null;
+    color: String | null;
+  }
+  const selectedOrgGroups = ref<OrgUserGroups[]>([]);
+
+  watch(orgDomainsData, (newOrgDomainsData) => {
+    if (newOrgDomainsData?.domainData) {
+      for (const domain of newOrgDomainsData.domainData) {
+        orgDomains.value.push({
+          domainPublicId: domain.publicId,
+          domain: domain.domain
+        });
+      }
+    }
+  });
 
   const formValid = computed(() => {
+    if (createEmailIdentityForGroup.value) {
+      if (isPro.value === false) {
+        return (
+          newIdentityUsernameValid.value === true &&
+          (newIdentitySendNameValid.value || newGroupNameValid.value) ===
+            true &&
+          selectedDomain.value?.domainPublicId &&
+          newGroupNameValid.value === true &&
+          !!newGroupColorValue.value
+        );
+      }
+      return (
+        newIdentityUsernameValid.value === true &&
+        (newIdentitySendNameValid.value || newGroupNameValid.value) === true &&
```

Comment:
Its not letting user just submit the form with already filed Send Name [ after filling group name ] initially its is group name !

Reply:
can we maybe check if the send name = temp name?
something like:
```
(
          newIdentityUsernameValid.value === true &&
          (newIdentitySendNameValid.value === true || (newIdentitySendNameTempValue.value === newIdentitySendNameValue.value) ===
            true &&
          selectedDomain.value?.domainPublicId &&
          newGroupNameValid.value === true &&
          !!newGroupColorValue.value
)
```

otherwise the groupname could be valid, but the send name could be invalid

Reply:
![uninboxSolveISSUE](https://github.com/uninbox/UnInbox/assets/132184385/1b6154a5-7e5a-4424-96a5-ec1b4dade7d6)

its giving another issue of not available

Reply:
created check function `validateSendName()` let me know if that works fine : )